src/librustc_resolve/lib.rs

```diff
@@ -3457,6 +3508,107 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         found_traits
     }
 
+    /// When name resolution fails, this method can be used to look up candidate
+    /// entities with the expected name. It allows filtering them using the
+    /// supplied predicate (which should be used to only accept the types of
+    /// definitions expected e.g. traits). The lookup spans across all crates.
+    ///
+    /// NOTE: The method does not look into imports, so it ignore all private-
+    /// made-public entities, such as the following code pattern:
+    /// ```
+    /// pub mod foo {
+    ///     mod bar { trait T {}; };
+    ///     // make private T accessible through an import:
+    ///     pub use bar::T as T;
+    /// }
```

Comment:
A couple things:
 - To `pub use` something, it has to be `pub`, so this example would be an error unless `trait T` was `pub`.
 - Since we don't care about accessibility in the current crate, we would report `foo::bar::T`, right?
 - `name_binding.is_import()` is can only be true for bindings in the current crate. Currently, our crate metadata does not distinguish between public imports and public items, so if this code were in an external crate, we would report `foo::T` as expected.